### ISSUE TITLE
TVRD() runs much slower in GDL than in IDL

### ISSUE THREAD

**Issue by Jeanne-jw** [gnudatalanguage/gdl#1266]
I have tried the function TVRD() in GDL and IDL on both Laptop Linux and  SSH remote server (Linux).
On both two machines the function runs faster in IDL than in GDL ( 5 ~ 15 times).
my_dustem_run_readme is 

### On Laptop Linux
In GDL on Laptop Linux :

```
.......
GDL> tic & R=TVRD(959,539,1,1) & toc
% Time elapsed :      0.012968063 seconds.
```

In IDL86 on Laptop Linux :

```
....... 
IDL> tic & R=TVRD(959,539,1,1) & toc 
% Time elapsed: 0.0021829605 seconds.
```

### On SSH remote server (Linux)
In GDL on SSH remote server (Linux):
```
......
GDL> tic & R=TVRD(959,539,1,1) & toc
% Time elapsed :       0.23119998 seconds.
```
In IDL88 on SSH remote server (Linux):
```
.......
IDL> tic & R=TVRD(959,539,1,1) & toc
% Time elapsed: 0.015527964 seconds.
```
In IDL86 on SSH remote server (Linux):
```
.......
IDL> tic & R=TVRD(959,539,1,1) & toc
% Time elapsed: 0.014930964 seconds.
```

**Comment by GillesDuvert** [comment]
This cannot be much improved alas.
The ratio of speed is even worse to tvscl().
This because we obey a complex architecture where pixels are not directly adressed by GDL, but through various intervening software levels (furthermore depending on the driver used).
I believe trying to improve this situation is an impasse given the current graphic architecture of gdl.

**Comment by GillesDuvert** [comment]
One side question is, why is 0.2 s a problem (or, equivalently, what applications would need IDL's 0.01 s)

**Comment by GillesDuvert** [comment]
On second thoughts, there is room for some speedup to help the compiler optimize the loops.
But your example is rather special, it copies only one pixel.
On a bona fide copy of he whole plot, IDL is not so good:
```
IDL> tic & R=TVRD() & toc       
% Time elapsed: 0.18007803 seconds.
IDL> help,r
R               BYTE      = Array[1720, 720]
```
```
GDL> tic & R=TVRD() & toc
% Time elapsed :      0.036110163 seconds.
GDL> help,r
R               BYTE      = Array[1720, 720]

```

**Comment by GillesDuvert** [comment]
I was mistaken. There was room for improvements, large ones in fact. See #1297 .
Runs now faster than IDL in all cases.

**Comment by GillesDuvert** [comment]
solved in #1297
